staging/src/k8s.io/client-go/examples/create-update-delete-deployment/main.go

```diff
@@ -97,35 +101,52 @@ func main() {
 	//    1. Modify the "deployment" variable and call: Update(deployment).
 	//       This works like the "kubectl replace" command and it overwrites/loses changes
 	//       made by other clients between you Create() and Update() the object.
-	//    2. Modify the "result" returned by Create()/Get() and retry Update(result) until
+	//    2. Modify the "result" returned by Get() and retry Update(result) until
 	//       you no longer get a conflict error. This way, you can preserve changes made
-	//       by other clients between Create() and Update(). This is implemented below:
-
-	for {
-		result.Spec.Replicas = int32Ptr(1)                    // reduce replica count
-		result.Spec.Template.Annotations = map[string]string{ // add annotations
-			"foo": "bar",
+	//       by other clients between Create() and Update(). This is implemented below
+	//			 using the retry utility package included with client-go. (RECOMMENDED)
+	//
+	// More Info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#concurrency-control-and-consistency
+
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
+		// Retrieve the latest version of Deployment before attempting update
+		// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
+		result, err = deploymentsClient.Get("demo-deployment", metav1.GetOptions{})
+		if err != nil {
+			panic(fmt.Errorf("Get failed: %+v", err))
 		}
 
-		if _, err := deploymentsClient.Update(result); errors.IsConflict(err) {
-			// Deployment is modified in the meanwhile, query the latest version
-			// and modify the retrieved object.
-			fmt.Println("encountered conflict, retrying")
-			result, err = deploymentsClient.Get("demo-deployment", metav1.GetOptions{})
-			if err != nil {
-				panic(fmt.Errorf("Get failed: %+v", err))
-			}
-		} else if err != nil {
-			panic(err)
-		} else {
-			break
+		result.Spec.Replicas = int32Ptr(1)                           // reduce replica count
+		result.Spec.Template.Spec.Containers[0].Image = "nginx:1.13" // change nginx version
+		result, err = deploymentsClient.Update(result)
+		return
+	})
+	if err != nil {
+		panic(fmt.Errorf("Update failed: %+v", err))
+	}
+	fmt.Println("Updated deployment...")
+
+	// Rollback Deployment
+	prompt()
+	fmt.Println("Rolling back deployment...")
+	// Once again use RetryOnConflict to avoid update conflicts
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
+		result, err = deploymentsClient.Get("demo-deployment", metav1.GetOptions{})
```

Comment:
this err, the err above on the left-side, and the err above in the anonymous function signature, is prone to scope issues. let's try to name them separately to avoid confusion

Reply:
good point, I should've caught that- I used the retry example without thinking.

will address that and do more thorough local testing, thanks for pointing that out!